### PR TITLE
Support Portainer EE deployment

### LINKED ISSUE #28: Support Portainer EE deployment
We need to enhance this repository to support the deployment of Portainer EE.

Todo:

- [x] Updated chart to support portainer EE deployment via `portainer/portainer-ee`
- [x] New manifests to deploy Portainer EE
- [x] Updated instructions with a section specific to Portainer EE

### PR DESCRIPTION
**Pull request by funkypenguin**
Fixes #28

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,9 +1,27 @@
-# Portainer Kubernetes Deployment
 
-This repo contains helm and YAML (yuch) for deploying Portainer into a Kubernetes environment
+This repo contains helm and YAML (yuch) for deploying Portainer into a Kubernetes environment. Follow the applicable instructions for your edition / deployment methodology below:
```
Comment (deviantony) [inline README.md:2]:
```suggestion
This repo contains helm and YAML for deploying Portainer into a Kubernetes environment. Follow the applicable instructions for your edition / deployment methodology below:
```

**Review by deviantony — CHANGES_REQUESTED** [review]
One small (yuch) change required :-)

**Review by deviantony — APPROVED** [review]
LGTM

File: deploy/manifests/portainer/portainer-ee.yaml [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+        app.kubernetes.io/instance: portainer
+    spec:
+      serviceAccountName: portainer-sa-clusteradmin
+      volumes:
+         - name: "data"
+           persistentVolumeClaim:
+             claimName: portainer      
+      containers:
+        - name: portainer
+          image: "portainer/portainer-ee:latest"
```
Comment (deviantony) [inline deploy/manifests/portainer/portainer-ee.yaml:111]: Shouldn't this one be pinned to 2.0.0 too?

File: deploy/manifests/agent/ee/portainer-agent-edge-k8s.yaml [hunk 3]
```diff
@@ -0,0 +1,95 @@
...
+      app: portainer-agent
+  template:
+    metadata:
+      labels:
+        app: portainer-agent
+    spec:
+      serviceAccountName: portainer-sa-clusteradmin
+      containers:
+      - name: portainer-agent
+        image: portainer/agent:latest
```
Comment (deviantony) [inline deploy/manifests/agent/ee/portainer-agent-edge-k8s.yaml]: Should be `portainer/agent:2.0.0`

File: deploy/manifests/portainer/portainer-lb-ee.yaml [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+        app.kubernetes.io/instance: portainer
+    spec:
+      serviceAccountName: portainer-sa-clusteradmin
+      volumes:
+         - name: "data"
+           persistentVolumeClaim:
+             claimName: portainer      
+      containers:
+        - name: portainer
+          image: "portainer/portainer-ee:latest"
```
Comment (deviantony) [inline deploy/manifests/portainer/portainer-lb-ee.yaml:109]: See my point above.

**Review by deviantony — CHANGES_REQUESTED** [review]
See my comments.

Also I was expecting all the files in https://github.com/portainer/k8s/tree/master/deploy/manifests/agent to have an equivalent in the EE folder?

File: deploy/manifests/agent/ee/portainer-agent-edge-k8s.yaml (hunk 3 above)
Comment (funkypenguin) [reply deploy/manifests/agent/ee/portainer-agent-edge-k8s.yaml]: Sorted ;)

File: deploy/manifests/portainer/portainer-lb-ee.yaml (hunk 4 above)
Comment (funkypenguin) [reply deploy/manifests/portainer/portainer-lb-ee.yaml]: Also sorted

File: deploy/manifests/agent/ee/portainer-edge-agent-setup.sh [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+    error "Not enough arguments"
+    error "Usage: ${0} <EDGE_ID> <EDGE_KEY>"
+    exit 1
+  fi
+
+  [[ "$(command -v curl)" ]] || errorAndExit "Unable to find curl binary. Please ensure curl is installed before running this script."
+  [[ "$(command -v kubectl)" ]] || errorAndExit "Unable to find kubectl binary. Please ensure kubectl is installed before running this script."
+
+  info "Downloading agent manifest..."
+  curl -L https://portainer.github.io/k8s/deploy/manifests/agent/portainer-agent-edge-k8s.yaml -o portainer-agent-edge-k8s.yaml || errorAndExit "Unable to download agent manifest"
```
Comment (deviantony) [inline deploy/manifests/agent/ee/portainer-edge-agent-setup.sh:58]: Should probably update that URL to point to the EE one.

**Review by deviantony — CHANGES_REQUESTED** [review]
One small thing to change

File: deploy/manifests/agent/ee/portainer-edge-agent-setup.sh (hunk 5 above)
Comment (funkypenguin) [reply deploy/manifests/agent/ee/portainer-edge-agent-setup.sh]: Right, missed that! Sorted now :)

**Review by deviantony — APPROVED** [review]
LGTM

Won't merge until EE is available.

**Pull request merged**